### DISCUSSION TITLE
How to do a full text search across all fileds in a collection?

### DISCUSSION THREAD

**Discussion by Mys7erio** [pocketbase/pocketbase#3807] [Q&A] [answered]
## How to do a full text search across all fileds in a collection?

I want to enable full text search so that I can query the entire database for a specific `search_term`. If the `search_term` is present in any field of a record, I want pb to return the record.

### Live Example
An excellent example of this is how pocketbase itself allows us to use a `searc_term` or a `search_filter`.

![Screenshot 2023-11-22 150049](https://github.com/pocketbase/pocketbase/assets/25553029/74785c47-c446-4cdd-a472-d6770702e9b6)

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase doesn't have support for fts at the moment (see https://github.com/pocketbase/pocketbase/issues/193).

The search in the Admin UI works by generating behind the scenes `~` filter rules for each field with the provided search term. For example:
```js
id ~ "searchTerm" || name ~ "searchTerm" || rollno ~ "searchTerm" || phoneno ~ "searchTerm" || priority ~ "searchTerm" || non_priority ~ "searchTerm" || discord ~ "searchTerm"
```

**Reply by ganigeorgiev** [reply]
If you use JS SDK v0.19.0+ or Dart SDK v0.16.0+, the above can be slightly simplified using the new binding parameters `pb.filter()` helper. Here is an example with the JS SDK:

```js
const searchTerm = "...";

await pb.collection("members").getList(1, 20, {
    filter: pb.filter(
        "id ~ {:search} || name ~ {:search} || rollno ~ {:search} || phoneno ~ {:search} || priority ~ {:search} || non_priority ~ {:search} || discord ~ {:search}",
        { search: searchTerm }
    )
})
```
But note that you still have to manually list all fields you want to wildcard search.

**Reply by Mys7erio** [reply]
Thanks a lot. I did end up implementing FTS this way.

**Reply by kilroyjones** [reply]
I have a query like this: 

```javascript
const movies = await locals.pocketbase.collection('movies').getList(1, 20, {
	filter: locals.pocketbase.filter('original_title ~ {:query}', { query: query })
});
```

Where query displays when printed to the console, but it returns no results. When I go the same **original_title ~  "<query>"** in the Pocketbase UI it works. Even when substituting in {:query} for a value that works in the UI, it still doesn't work. Any idea?

**Reply by ganigeorgiev** [reply]
@kilroyjones make sure that your client request satisfies the List API rule of the `movies` collection (API rules don't apply for admins so that would explain why it works from the Admin UI).

If the above doesn't help, please open a new Q&A discussion with more details about your collections structure (Admin UI > Settings > Export collections).

**Reply by kilroyjones** [reply]
I just came back to remove this post after realizing I hadn't properly set the API rules. Thanks for the speedy reply.
